src/Qowaiv.Bson.MongoDB/Qowaiv.Internals/ConventionBasedSerializer.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using System.Globalization;
+using MongoDB.Bson;
+using MongoDB.Bson.Serialization;
+using MongoDB.Bson.Serialization.Serializers;
+
+namespace Qowaiv.Internals
+{
+    internal partial class ConventionBasedSerializer<TSvo> : SerializerBase<TSvo>
+    {
+        /// <inheritdoc/>
+        public override TSvo Deserialize(BsonDeserializationContext context, BsonDeserializationArgs args)
+        {
+            Guard.NotNull(context, nameof(context));
+
+            var reader = context.Reader;
+            var bsonType = reader.GetCurrentBsonType();
+
+            switch (bsonType)
+            {
+                case BsonType.Null:
+                    return default;
+
+                case BsonType.DateTime:
+                case BsonType.String:
+                    return FromJson(reader.ReadString());
+
+                case BsonType.Double:
+                    return FromJson(reader.ReadDouble());
+
+                case BsonType.Boolean:
+                    return FromJson(reader.ReadBoolean());
+
+                case BsonType.Int32:
+                    return FromJson(reader.ReadInt32());
+
+                case BsonType.Int64:
+                    return FromJson(reader.ReadInt64());
+
+                // These values are not supported:
+                //
+                // case BsonType.Decimal128:
+                // case BsonType.EndOfDocument:
+                // case BsonType.Document:
+                // case BsonType.Array:
+                // case BsonType.Binary:
+                // case BsonType.Undefined:
+                // case BsonType.ObjectId:
+                // case BsonType.RegularExpression:
+                // case BsonType.JavaScript:
+                // case BsonType.Symbol:
+                // case BsonType.JavaScriptWithScope:
+                // case BsonType.Timestamp:
+                // case BsonType.MinKey:
+                // case BsonType.MaxKey:
+                default:
+                    throw CreateCannotDeserializeFromBsonTypeException(reader.GetCurrentBsonType());
+            }
+        }
+
+        /// <inheritdoc/>
+        public override void Serialize(BsonSerializationContext context, BsonSerializationArgs args, TSvo value)
+        {
+            Guard.NotNull(context, nameof(context));
+
+            var writer = context.Writer;
+            var obj = ToJson(value);
+
+            if (obj is null)
+            {
+                writer.WriteNull();
+            }
+            else if (obj is string str)
+            {
+                writer.WriteString(str);
+            }
+            else if (obj is decimal dec)
+            {
+                writer.WriteDouble((double)dec);
+            }
+            else if (obj is double dbl)
+            {
+                writer.WriteDouble(dbl);
+            }
```

Comment:
`else if (obj is float fl){ .. }`
`else if (obj is short fl){ .. }`
`else if (obj is byte fl){ .. }`

Reply:
Those options are also not in the others. The idea is that there are not usecases (yet) to communcate these types as ToJson(), as it are no JSON types. But you think ToJson() should be able to return these types?

Reply:
Ok, I get it.